master/internal/job/job.go

```diff
@@ -133,6 +141,213 @@ func (j *Jobs) jobQSnapshot(ctx *actor.Context, resourcePool string) (AQueue, er
 	return jobQ, nil
 }
 
+// figure out what changes are needed to move the job to the desired position.
+// Later we can support cross RP moving this way as well.
+func moveJobMessages(
+	jobs []*jobv1.Job,
+	target *jobv1.Job,
+	anchor *jobv1.Job,
+	anchorIdx int,
+	aheadOf bool,
+) (*SetGroupPriority, *MoveJob, error) {
+	// validate anchorIdx
+	if anchorIdx < 0 || anchorIdx > len(jobs) {
+		return nil, nil, fmt.Errorf("invalid anchor index %d", anchorIdx)
+	}
+	// validate anchor and target
+	if anchor == nil {
+		return nil, nil, fmt.Errorf("missing anchor job")
+	}
+	if target == nil {
+		return nil, nil, fmt.Errorf("missing target job")
+	}
+	if target.JobId == anchor.JobId {
+		return nil, nil, nil
```

Comment:
maybe raise error? Target job cannot be the same as the anchor? Or target cannot refer to itself?

Reply:
I was gonna say it could be fine without an error but the fact that the user has specified either an ahead of or a behind of makes it so that it has to be an error.